objects/src/assets/nonfungible.rs

```diff
@@ -92,12 +97,9 @@ impl NonFungibleAsset {
         let faucet_id = AccountId::try_from(self.0[FAUCET_ID_POS])
             .map_err(|e| AssetError::InvalidAccountId(e.to_string()))?;
 
-        if !matches!(faucet_id.account_type(), AccountType::NonFungibleFaucet) {
-            return Err(AssetError::not_a_fungible_faucet_id(faucet_id));
-        }
-
-        if self.0[3].as_int() >> 63 != 0 {
```

Comment:
There were quite a few places that used these magic constants. Whenever possible I tried to:

- Use a method from the account_id, since that hides the format
- If that was not possible, then I changed the code to use proper constants, instead of magic values